models/PostModel.js

```diff
@@ -156,10 +167,23 @@ const getReplies = `
     ON p."postId" = r."parentPostId"
   INNER JOIN app_user AS u
     ON p."userId" = u."userId"
+  LEFT JOIN post AS parent_post
+    ON p."parentPostId" = parent_post."postId"
   WHERE p."parentPostId" = $2
+    AND p."deleted" = FALSE
   ORDER BY "numberOfLikes" DESC, "timestamp" DESC;
 `;
 
+const deletePost = `WITH post_to_delete AS (
+  SELECT "postId"
+  FROM post
+  WHERE "postId" = $1 AND "userId" = $2 AND deleted = FALSE
+)
+UPDATE post
+SET deleted = TRUE
+WHERE "postId" IN (SELECT "postId" FROM post_to_delete)
+RETURNING "postId", deleted;`;
```

Comment:
What does the cte do? Can we not just perform and UPDATE clause? I don't think we need the user id either, because that is just to validate that they can delete the post. Once we know that they have permission to delete the post, we only need the postId to query by something like:
```suggestion
const deletePost = ` 
  UPDATE posts
  SET deleted = TRUE
  WHERE "postId" = $1
    AND "deleted" = FALSE
  RETURNING "postId", deleted;
  `;

```

Reply:
The CTE was purely for validation of the user but since this check is being handled on the frontend you are right we don't need this.

Reply:
Yeah, I agree that in real life we should still check that the user is able to delete this post on the server side. My thinking was that this security logic more belongs in the controller rather than the query though. Like I think we should go with your idea of checking the user's id using the jwt token and confirm that they're able to delete the post there

Let's say we check for the user'd id here and find out it doesn't match, then nothing would happen, so we wouldn't really be able to throw an error the same way we could in the controller. If we check in the controller, then we don't need this user id check, we can just straight delete the post using the post id. Let me know your thoughts on this

Reply:
That makes more sense to me now, I can get that going for your review right away. This kind of verification does belong in the controller rather than the query itself